### PR TITLE
Clevo nv41mz/persistent bootlogo

### PR DESCRIPTION
**Pull request by mkopec**

### CODE REVIEW

File: src/mainboard/clevo/tgl-u/vboot-rwa.fmd [hunk 1]
```diff
@@ -6,7 +6,7 @@ FLASH@0xff000000 0x1000000 {
 
 	RW_SECTION_A 0x400000 {
```
Comment (miczyg1) [inline src/mainboard/clevo/tgl-u/vboot-rwa.fmd:7]: We need a few adjustments here:
1. Place RW_MISC as first region and define its size (it is fixed so it won't change I believe)
2. Place bootsplash and console as next after RW_MISC
3. Place RW_SECTION_A as last just before WP_RO and do not define the size of RW_SECTION_A, let it stretch to whole free space.

**Review by miczyg1 — CHANGES_REQUESTED** [review]
Please fix the comments

File: src/mainboard/clevo/tgl-u/vboot-rwa.fmd (hunk 1 above)
Comment (mkopec) [reply src/mainboard/clevo/tgl-u/vboot-rwa.fmd]: Applied the suggestions. Tested, boots correctly.

File: src/lib/coreboot_table.c [hunk 2]
```diff
@@ -388,6 +390,50 @@ void __weak lb_board(struct lb_header *header) { /* NOOP */ }
...
+	struct region_device rdev;
+	struct bootlogo_header header;
+	void *logo_buffer;
+
+	struct cbfsf logo_file;
+	uint32_t compression;
+	uint32_t cbfs_type = CBFS_TYPE_RAW;
+	size_t logo_size;
+
+	logo_entry = cbmem_entry_add(CBMEM_ID_TIANOCORE_LOGO, 4 * MiB);
```
Comment (miczyg1) [inline src/lib/coreboot_table.c:409]: Why don't we create the memory buffer only as big as the logo + header with size? It is a waste of memory

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: src/lib/coreboot_table.c (hunk 2 above)
Comment (mkopec) [reply src/lib/coreboot_table.c:409]: The buffer in cbmem is larger to accomodate uncompressed bitmaps. The file in CBFS may be up to 1MB compressed and up to 4 MB uncompressed.

Comment (miczyg1) [reply src/lib/coreboot_table.c:409]: I understand it pretty clearly, but we can create cbmem entry after the bitmap is loaded and uncompressed, note before. That way we may alloc just the amount of memory we need.

Comment (mkopec) [reply src/lib/coreboot_table.c]: Right. Updated as per suggestion

File: build.sh [hunk 3]
```diff
@@ -30,38 +30,18 @@ FW_FILE="dasharo_${BOARD}_${FW_VERSION}.rom"
 HASH_FILE="${FW_FILE}.SHA256"
 SIG_FILE="${HASH_FILE}.sig"
 ARTIFACTS_DIR="artifacts"
-LOGO=""
+LOGO="3rdparty/dasharo-blobs/novacustom/bootsplash.bmp"
 
 [ -z "$FW_VERSION" ] && errorExit "Failed to get FW_VERSION - CONFIG_LOCALVERSION is probably not set"
 
-replace_logo() {
```
Comment (miczyg1) [inline build.sh:37]: Why do we remove it?

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: src/lib/coreboot_table.c [hunk 4]
```diff
@@ -533,6 +579,8 @@ void *write_tables(void)
 	size_t cbtable_size;
 	const size_t max_table_size = COREBOOT_TABLE_SIZE;
 
+	tianocore_logo_load(); // wrong place for this
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/lib/coreboot_table.c]: Added as a CBMEM hook (`RAMSTAGE_CBMEM_INIT_HOOK`)

File: src/commonlib/include/commonlib/coreboot_tables.h [hunk 5]
```diff
@@ -438,6 +439,13 @@ struct lb_board_config {
 	uint32_t sku_id;
 };
 
+struct lb_logo {
```
Comment (miczyg1) [inline src/commonlib/include/commonlib/coreboot_tables.h:442]: unused?

**Review by miczyg1 — CHANGES_REQUESTED** [review]

**Comment by mkopec** [comment]
@miczyg1 is there anything left to do here?

**Review by miczyg1 — APPROVED** [review]

**Pull request merged**